client/src/error.rs

```diff
@@ -13,22 +13,40 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use failure::{Compat, Fail};
 use jsonrpc_core_client::RpcError;
 use parity_scale_codec::Error as CodecError;
+use thiserror::Error;
```

Comment:
following to the discussion around ambiguity, I'd suggest us applying that here already:

```suggestion
use thiserror::Error as ThisError;
```

Reply:
Interesting case of ambiguity, unclear for humans, but easy for the compiler, good catch! I'd rather avoid aliasing if not necessary.

Fixed, but differently.

Reply:
ahah indeed, it;s usually the way around. Your fix is good :+1: